django/db/models/base.py

```diff
@@ -811,11 +811,16 @@ def save(
             if loaded_fields:
                 update_fields = frozenset(loaded_fields)
 
+        calling_frame = inspect.currentframe().f_back
+        calling_function_name = calling_frame.f_code.co_name
+        for_create = calling_function_name == "create"
```

Comment:
This is very hacky. Unfortunately, we cannot accept such a solution because inspecting the stack frame is very expensive and would introduce performance regression, it's also unreadable, tricky, and error-prone.

I'm pretty sure there is an alternative solution for this, perhaps an extra layer/hook that will allow appropriate flags to be passed in a convenient way.